src/utils/wallet.js

```diff
@@ -143,9 +157,25 @@ const wallet = {
       /* eslint-disable no-await-in-loop */
       const balance = await wallet.getBalance(token);
       const tokenBalance = balance[0].balance;
-      tokensBalance[token] = { available: tokenBalance.unlocked, locked: tokenBalance.locked };
+      const authorities = balance[0].tokenAuthorities;
+
+      let mint = false;
+      let melt = false;
+
+      if (authorities) {
+        const { unlocked } = authorities;
+        mint = unlocked.mint;
+        melt = unlocked.melt;
+      }
+
+      tokensBalance[token] = {
+        available: tokenBalance.unlocked,
+        locked: tokenBalance.locked,
+        mint,
+        melt,
+      };
       // We fetch history count of 5 pages and then we fetch a new page each 'Next' button clicked
-      const history = await wallet.getTxHistory({ token_id: token, count: 5 * WALLET_HISTORY_COUNT });
+      const history = await wallet.getTxHistory({ token_id: token, count: WALLET_HISTORY_COUNT });
```

Comment:
I did that because the wallet service has a limit of 15 transactions per request

The history pagination is not working on this branch, but I suggest fixing it on another PR so we can approve this one.

What do you think? I won't merge this one on `dev` until that new PR is done and reviewed

Reply:
We can fix this in another PR. Did you create this issue and added to the project so we don't consider the project done with this opened?

While we don't fix what's happening with the pagination? 

You should at least remove the comment that is wrong now.

Reply:
Created here: https://github.com/HathorNetwork/hathor-wallet/issues/303

And added it to the project